api_server/src/http_service.rs

```diff
@@ -537,11 +545,19 @@ impl hyper::server::Service for ApiServerHttpService {
                         // Requests on /mmds should not have the body in the logs as the data
                         // store contains customer data.
                         log_received_api_request(describe(&method_copy, &path, &None));
-                        mmds_info
+                        let mut mmds = mmds_info
                             .lock()
-                            .expect("Failed to acquire lock on MMDS info")
-                            .put_data(json_value);
-                        Either::A(future::ok(empty_response(StatusCode::NoContent)))
+                            .expect("Failed to acquire lock on MMDS info");
+                        if mmds.put_data(json_value).is_ok() {
```

Comment:
Same here - transform `Result`s with [`map`](https://doc.rust-lang.org/std/result/enum.Result.html#method.map) and [`map_err`](https://doc.rust-lang.org/std/result/enum.Result.html#method.map_err).